packages/denylist/src/cache.js

```diff
@@ -0,0 +1,38 @@
+/* eslint-env browser */
+/**
+ * Intercepts request if content cached by just returning cached response.
+ * Otherwise proceeds to handler.
+ * @param {import('itty-router').RouteHandler<Request>} handler
+ */
+export function withCdnCache (handler) {
+  /**
+   * @param {Request} request
+   * @param {import('./env').Env} env
+   * @param {ExecutionContext} ctx
+   */
+  return async (request, env, ctx) => {
+    // Should skip cache if instructed by headers
+    if ((request.headers.get('Cache-Control') || '').includes('no-cache')) {
+      return handler(request, env, ctx)
+    }
+
+    let response
+    // Get from cache and return if existent
+    /** @type {Cache} */
+    // @ts-ignore Cloudflare Workers runtime exposes a single global cache object.
```

Comment:
```suggestion
    // @ts-expect-error Cloudflare Workers runtime exposes a single global cache object.
```

we have been trying to use expect error instead of ignore. This is cool because if things get fixed in the meantime (e.g CF gets its types to get this in, we would get an error that would trigger us to remove the expect-error line :)